### PR TITLE
Add some comments and renames to client code

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: client/client-lib/src/lib.rs [hunk 1]
```diff
@@ -214,32 +215,36 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
...
+    /// Validate signatures on notes.
+    ///
+    /// This function checks if signatures are valid
     /// based on the federation public key. It does not check if the nonce is unspent.
-    pub async fn validate_tokens(&self, coins: &TieredMulti<SpendableNote>) -> Result<()> {
+    pub async fn validate_note_signatures(&self, notes: &TieredMulti<SpendableNote>) -> Result<()> {
         let tbs_pks = &self.mint_client().config.tbs_pks;
-        coins.iter_items().try_for_each(|(amt, coin)| {
+        notes.iter_items().try_for_each(|(amt, coin)| {
             if coin.coin.verify(*tbs_pks.tier(&amt)?) {
```
Comment (justinmoon) [inline client/client-lib/src/lib.rs:248]: Should probably rename these `coin` variables ...

File: client/client-lib/src/mint/mod.rs [hunk 2]
```diff
@@ -44,12 +46,14 @@ pub struct CoinRequest {
...
 
-/// Client side representation of a coin reissuance that keeps all necessary information to
-/// generate spendable coins once the blind signatures arrive.
+/// Multiple [`Note`] issuance requests
+///
+/// Keeps all the data to generate [`SpendableNote`]s once the
+/// mint successfully processed corresponding [`NoteIssuanceRequest`]s.
 #[derive(Debug, Clone, Deserialize, Serialize, Encodable, Decodable)]
-pub struct CoinFinalizationData {
+pub struct IssuanceRequests {
```
Comment (justinmoon) [inline client/client-lib/src/mint/mod.rs:54]: `NoteIssuanceRequests` just to be consistant?

File: client/client-lib/src/transaction.rs [hunk 3]
```diff
@@ -14,7 +14,7 @@ use tbs::AggregatePublicKey;
 
 pub struct TransactionBuilder {
     input_coins: TieredMulti<SpendableNote>,
-    output_coins: Vec<(u64, CoinFinalizationData)>,
+    output_coins: Vec<(u64, IssuanceRequests)>,
```
Comment (justinmoon) [inline client/client-lib/src/transaction.rs:17]: coins -> note

**Comment by justinmoon** [comment]
I think these are good changes. Try to finish them off!

**Comment by dpc** [comment]
> I think these are good changes. Try to finish them off!

@justinmoon Done.

**Review by elsirion — APPROVED** [review]

**Pull request merged**